### ISSUE TITLE
Should we ensure no name generation is required?

### ISSUE THREAD

**Issue by Pyrolistical** [WebAssembly/module-linking#43]
Right now it's impossible to write an alias that is round-trippable.

The issue is `alias-name ::= (func <funcidx>) | ...`. If `funcidx` were made optional, then we could write,
```wasm
(alias 0 "f" (func))
(instance (instantiate 0 (import "foo" (func 0))))
```

Without this, we are currently forced to generate a name
```wasm
(alias 0 "f" (func $generated_alias))
(instance (instantiate 0 (import "foo" (func $generated_alias))))
```

Does core wasm require generation of names anywhere? If not, I think we should retain that behaviour.

**Comment by lukewagner** [comment]
Good point!  I think this is actually a typo in the Explainer.md.  Instead of saying `<funcidx>`, it should just say `<id>?`, with the identifier being a purely text-format affordance.

**Comment by Pyrolistical** [comment]
Would this also apply to this line? `type ::= (type $id <deftype>)` where `$id` should also be optional?

**Comment by lukewagner** [comment]
Yep!

**Comment by lukewagner** [comment]
Resolved by #44.
